src/ufo.rs

```diff
@@ -205,12 +205,34 @@ impl Ufo {
                 (_, None, k) => (None, k), // Without a groups.plist, there's nothing to upgrade.
                 (_, Some(g), k) => {
                     let (groups, kerning) =
-                        upconvert_kerning(&g, &k.unwrap_or_default(), &glyph_names);
+                        upconversion::upconvert_kerning(&g, &k.unwrap_or_default(), &glyph_names);
                     validate_groups(&groups).map_err(Error::GroupsUpconversionError)?;
                     (Some(groups), Some(kerning))
                 }
             };
 
+            // The v1 format stores some Postscript hinting related data in the lib,
+            // which we only import into fontinfo if we're reading a v1 UFO.
+            if meta.format_version == FormatVersion::V1 {
+                let mut feature_text = String::new();
+                let mut fontinfo =
+                    if let Some(fontinfo) = font_info { fontinfo } else { FontInfo::default() };
+
+                if let Some(lib_data) = &mut lib {
+                    upconversion::upconvert_ufov1_robofab_data(
+                        &lib_path,
+                        lib_data,
+                        &mut feature_text,
```

Comment:
I don't think it makes sense to have this be an inout argument; I'd prefer to just return a tuple from this function.